File: Data Structures & Algorithms/balanced-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isBalanced(TreeNode root) {
       if (root == null){
        return true;
       }
       else{
        int ls = no(root.left);
        int rs = no(root.right);

        int diff = ls - rs;

        if(Math.abs(diff) >1){
            return false;
        }
        boolean lft = isBalanced(root.left);
        boolean rft = isBalanced(root.right);
        return lft && rft;
       }
       
        }
    private int no (TreeNode root){
            if (root == null){
                return 0;
            }
            else{
                int ls = no(root.left);
                int rs = no(root.right);
                return 1 + Math.max(ls,rs);
            }
        }
    }

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.goodnodes = 0

        def good(root, maxval):
            if not root:
                return None

            if root.val >= maxval:
                self.goodnodes += 1
                
            maxval = max(maxval,root.val)

            if root.left:
                good(root.left,maxval)

            if root.right:
                good(root.right,maxval)
           
           

        good(root, -float("inf"))

        return self.goodnodes
